perses/rjmc/topology_proposal.py

```diff
@@ -1470,6 +1474,98 @@ def generate_oemol_from_pdb_template(pdbfile):
             atom.SetName(name_without_spaces)
         return current_oemol
 
+    @staticmethod
+    def _get_charge_difference(current_resname, new_resname):
+        """
+        return the charge of the new res - charge old res
+
+        Parameters
+        ----------
+        current_resname : str
+            three letter identifier for original residue
+        new_resnme : str
+            three letter identifier for new residue
+
+        Returns
+        -------
+        chargediff : int
+            charge(new_res) - charge(old_res)
+        """
+        assert new_resname in PolymerProposalEngine._aminos
+        assert current_resname in PolymerProposalEngine._aminos
+
+        new_rescharge, current_rescharge = 0,0
+        resname_to_charge = {current_resname: 0, new_resname: 0}
+        for resname in [new_resname, current_resname]:
+            if resname in PolymerProposalEngine._negative_aminos:
```

Comment:
How do we treat residues in different protonation states? Do we require that we use `HIS`, `HID`, `HIE` names to access different protonation states of histidine, for example? I know OpenMM uses PDB standard names, not AMBER protonation state variant names.

Is there a more robust way to determine the net charge on the system, such as using the total partial charges from the force field from the unmodified system? That would presumably be more robust, even if you just have to round the fractional charges.

Reply:
>How do we treat residues in different protonation states? Do we require that we use HIS, HID, HIE names to access different protonation states of histidine, for example?

It's on my TODO list to fix how we handle protonation states in Perses. Currently, I think we always choose `HIS`. 

>Is there a more robust way to determine the net charge on the system

This would be more robust, which would be useful for non-protein systems. I'll open an issue for us to address that later.